miden-lib/asm/kernels/transaction/main.masm

```diff
@@ -159,11 +158,14 @@ proc.main.1
 
     # execute the transaction epilogue
     exec.epilogue::finalize_transaction
-    # => [OUTPUT_NOTES_COMMITMENT, FINAL_ACCOUNT_HASH]
+    # => [OUTPUT_NOTES_COMMITMENT, FINAL_ACCOUNT_HASH, tx_expiration_block_num]
+
+    # truncate the stack
+    movupw.3 dropw movupw.3 dropw movup.9 drop
```

Comment:
I think that's a good solution for https://github.com/0xPolygonMiden/miden-base/issues/902.

So we can close that issue once this PR is merged.